Sage/carbon
# fix(filterable-select): ensure selectionConfirmed is false when enter pressed with no match

### Proposed behaviour

When typing text into a FilterableSelect input that does not match any options, pressing Enter should clear the input and not set the `selectionConfirmed` property to true in the event object passed to the `onChange` callback.

fix #6518 



### Current behaviour

If there was previously a match and a new character is typed that results in  no mathes, pressing Enter results in `onChange` being called with the `selectionConfirmed` property on the event set to true.



### Checklist




#### QA


### Additional context



### Testing instructions


Test against the codesandbox in the issue, to ensure the problem is fixed. (Can also be tested against the "selectionConfirmed" test story under FilterableSelect.

Perform full regression testing against all 3 types of Select to ensure everything else is still behaving as intended.
